src/CompoundTypeInterface.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+namespace Dhii\Type;
+
+/**
+ * Represents a compound type.
+ *
+ * @since [*next-version*]
+ */
+interface CompoundTypeInterface extends TypeInterface
```

Comment:
Consider different naming, such as `StructureTypeInterface`, since "compound" means made up of two or more elements, which is not a restriction for these types.